app/src/main/java/org/mozilla/focus/search/SearchEngineManager.java

```diff
@@ -231,20 +235,29 @@ private synchronized void loadFromDisk(Context context) {
     }
 
     private List<SearchEngine> loadCustomSearchEngines(Context context) {
-        final List<SearchEngine> searchEngines = new LinkedList<>();
+        final List<SearchEngine> customEngines = new LinkedList<>();
         final SharedPreferences prefs = context.getSharedPreferences(PREF_FILE_SEARCH_ENGINES, Context.MODE_PRIVATE);
         final Set<String> engines = prefs.getStringSet(PREF_KEY_CUSTOM_SEARCH_ENGINES, Collections.<String>emptySet());
         try {
             for (String engine : engines) {
                 final InputStream engineInputStream = new ByteArrayInputStream(prefs.getString(engine, "").getBytes(StandardCharsets.UTF_8));
-                searchEngines.add(SearchEngineParser.load(engine, engineInputStream));
+                customEngines.add(SearchEngineParser.load(engine, engineInputStream));
             }
         } catch (IOException e) {
             Log.e(LOG_TAG, "IOException while loading custom search engines", e);
         } catch (XmlPullParserException e) {
             Log.e(LOG_TAG, "Couldn't load custom search engines", e);
         }
-        return searchEngines;
+        return customEngines;
+    }
+
+    public boolean isCustomSearchEngine(String engineId, Context context) {
+        for (SearchEngine e : loadCustomSearchEngines(context)) {
```

Comment:
Mhh. This will always re-parse all custom engines. Not super nice - but then again it is also unlikely that a user has a huge list of custom engines.

Reply:
Yeah, I'm not excited about this either - my first version cached this, but this late in the sprint I wanted to minimize the complexity of making sure this is non-null or up-to-date.